standard/lpdb/count.lua

```diff
@@ -0,0 +1,274 @@
+function Count.tournaments(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args, true)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	return Count._query('tournament', lpdbConditions)
+end
+
+
+---Counts the number of placements for a specified opponent on a wiki
+---@param args table
+---@return integer?
+function Count.placements(args)
+	args = args or {}
+
+	local lpdbConditions = Count._baseConditions(args)
+	lpdbConditions = Count._tierConditions(args, lpdbConditions)
+
+	if not Logic.readBool(args.includeShowmatch) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Showmatch')}
+	end
+
+	if not Logic.readBool(args.includeQualifier) then
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatier'), Comparator.neq, 'Qualifier')}
+		lpdbConditions:add{ConditionNode(ColumnName('liquipediatiertype'), Comparator.neq, 'Qualifier')}
+	end
+
+	if Logic.readBool(args.noEmptyPrizePool) then
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.neq, '')}
+		lpdbConditions:add{ConditionNode(ColumnName('prizemoney'), Comparator.gt, 0)}
+	end
+
+	if String.isNotEmpty(args.player) then
+		local opponent = mw.ext.TeamLiquidIntegration.resolve_redirect(args.player)
+		local opponentWithUnderscores = opponent:gsub(' ', '_')
+		local opponentConditions = ConditionTree(BooleanOperator.any)
+		for index = 1, 10 do
+			opponentConditions:add{
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponent),
+				ConditionNode(ColumnName('opponentplayers_p' .. index), Comparator.eq, opponentWithUnderscores)
+			}
+		end
+		lpdbConditions:add{opponentConditions}
+
+	elseif String.isNotEmpty(args.team) then
+		local opponentConditions = ConditionTree(BooleanOperator.any)
+		Array.map(Count._getOpponentNames(args.team),
+		function(templateValue)
+			return opponentConditions:add{
+				ConditionNode(ColumnName('opponentname'), Comparator.eq, templateValue),
+				ConditionNode(ColumnName('opponentname'), Comparator.eq, templateValue:gsub(' ', '_'))
+			}
+			end
+		)
+		lpdbConditions:add{opponentConditions}
+	end
+
+	if String.isNotEmpty(args.placement) then
+		local placementConditions = ConditionTree(BooleanOperator.any)
+		Array.map(Array.map(mw.text.split(args.placement, ',', true), String.trim),
+		function(placementValue)
+			return placementConditions:add{
+				ConditionNode(ColumnName('placement'), Comparator.eq, placementValue)}
+			end)
+		lpdbConditions:add{placementConditions}
+	end
+
+	return Count._query('placement', lpdbConditions)
+end
+
+
+---Returns the counted number based on the type of query
+---@param queryType string
+---@param lpdbConditions ConditionTree
+---@return integer?
+function Count._query(queryType, lpdbConditions)
+	local data = mw.ext.LiquipediaDB.lpdb(queryType, {
+		conditions = lpdbConditions:toString(),
+		query = 'count::objectname',
+	})
+
+	return tonumber(data[1]['count_objectname'])
```

Comment:
doesn't this return 0 if it can't find results?
i.e. annos would be without `?` and you would need a `--[[@as integer]]` here

Reply:
it tells the annos that the var is an integer now